modules/default.nix

```diff
@@ -33,7 +33,9 @@ let
     modules = [ configuration ] ++ hmModules;
     specialArgs = {
       modulesPath = builtins.toString ./.;
-    } // extraSpecialArgs;
+    } // extraSpecialArgs // optionalAttrs (extraSpecialArgs ? lib) {
+      lib = extendedLib;
+    };
```

Comment:
I'm also not sure why this is needed: the module system should take care of passing the correct `lib` as a "special" argument to every module.

Reply:
Redundant question, comes down to the same question in your issue comment, thus see my response to that.